src/properties/propertiesService.ts

```diff
@@ -27,6 +29,38 @@ interface PropertySchemas {
   uiSchema?: UISchemaElement;
 }
 
+const ruleSchema = {
+  rule: {
```

Comment:
can't we be more precise with the schema? 
Do we need to be more precise?

Reply:
No, not at this point. Changing the schema here has no effect on the editor and if any of the mandatory properties is missing (e.g. `effect`) JSON Forms will crash either way. I think this can be handled by giving the editor a schema to validate against, like you suggested above.